Volume roll current contracts could flip back and forth (Revised)

dmichalowicz opened the pull request:
See https://github.com/quantopian/zipline/pull/1915 for description.

dmichalowicz commented on zipline/assets/roll_finder.py:
```diff
@@ -226,3 +238,38 @@ def _active_contract(self, oc, front, back, dt):
+        end_date = min(
+            dt + (ROLL_DAYS_FOR_CURRENT_CONTRACT * day),
+            self.session_reader.last_available_dt,
+        )
```
@yankees714 I reopened this PR from the changes I reverted, this is the only piece I changed as I discussed with you this morning. (previously this was just `dt + (ROLL_DAYS_FOR_CURRENT_CONTRACT * day)`). And I added a short test.

yankees714 commented on tests/test_continuous_futures.py:
```diff
@@ -1523,6 +1523,35 @@ def test_end_before_auto_close(self):
+            self.asset_finder.retrieve_asset(1003),
+        )
+        self.assertEqual(
+            get_contract_center(dt=pd.Timestamp('2017-04-18', tz='UTC')),
```
I'm thinking it might be more maintainable to define these dates relative to `self.END_DATE`, i.e. in https://github.com/quantopian/zipline/pull/1917 I pushed the end date back, which would have broken the intention of these tests and I don't think I would have realized it.

dmichalowicz commented:
@yankees714 I updated my test based on your suggestion. To make it easier I just added a new contract chain entirely. If this looks good I'll wait to merge until you merge #1917 because I'll probably have to resolve rebase conflicts.
